### ISSUE TITLE
when target of an `f` operation is `(` or `)` cursor doesn't jump to it (or if it does it's invisible)

### ISSUE THREAD

**Issue by alexandreroberts** [ggandor/flit.nvim#44]
When I type `f(` the cursor does not appear to jump to the next `(`. But it clearly has because its pair is highlighted. E.g., if the cursor is on the first column in this line

```
word word (lkjsdf)
```

and then I type `(`, the highlighting only changes for the `)`, not for the `(`, because I am using a package that highlights matching pairs. In any case, the `(` should be highlighted to show that the cursor has arrived.

Perhaps this is related to [this leap issue](https://github.com/ggandor/leap.nvim/issues/70)?

**Comment by ggandor** [comment]
> Perhaps this is related to https://github.com/ggandor/leap.nvim/issues/70?

Yep, might be related. So it's not a problem with the `MatchParen` highlight, if you just move the cursor to `(`, it is visible, right?

**Comment by alexandreroberts** [comment]
Yes, that's right: when I simply move the cursor to `(`, in this case by typing `ww`, it is highlighted as normally (and also its matching paren is highlighted) -- exactly the expected behavior.

I should add that the issue seems to be restricted to whatever mode I'm in while using flit's `f`, where all the possible targets are marked. I say this because in the above example (`word word (lkjsdf)`), if I type `f(<esc>`, then the expected highlighting reappears, as if I had moved there with `ww`. So the problem exists only between the time I enter flit's `f` mode (by typing `f(`) and the time I exit that mode, in this case by entering normal mode with the escape key.

**Comment by alexandreroberts** [comment]
I have another observation to add: this problem takes place when flit's target is `(` or `{`, but *not* when the target is `[`.

Does that help narrow down what is causing the problem?

(By the way, I just updated to Neovim 0.9.5, and the problem persists.)
